src/lambda.ts

```diff
@@ -48,7 +55,9 @@ export class WatchLambdaFunction extends Construct {
 
     const cfnFunction = props.fn.node.defaultChild as lambda.CfnFunction;
     // if we have an IFunction or an unspecified timeout, use the default of 3 seconds
-    const timeoutSec = cfnFunction? (cfnFunction.timeout? cfnFunction.timeout : 3) : 3;
+    let timeoutSec = cfnFunction? (cfnFunction.timeout? cfnFunction.timeout : 3) : 3;
+    // if a threshold was passed for duration use it
+    timeoutSec = props.durationTimeout? props.durationTimeout: timeoutSec;
```

Comment:
this is a bit rough to read. maybe time for an `if`? 
Also probably worth checking if the prop exists first instead of doing that last.

Reply:
unfortunately, we have to have a path where we must assign a value, or else `cdk deploy` will complain with `'null' values are not allowed in templates`.  for that reason, i think this should come last

Reply:
null should never get assigned as a value, it should default to 3 when the other paths fail. Happy to pair if you want to flesh out the coalescence